abcclassroom/github.py

```diff
@@ -182,6 +307,27 @@ def init_and_commit(directory, custom_message=False):
         print("No changes to local repository.")
 
 
+def _master_branch_to_main(directory):
+    """Change the name of the master branch to main
+
+    Changes the name of the master branch to main for the repo in the
+    given directory. Since we create the repo on github first, which now sets
+    the default branch to 'main', we need the local repo to match
+    in order to be able to push with error later.
+    """
+    print(
+        """Changing name of 'master' branch to 'main'
+        in repo {}""".format(
+            directory
+        )
+    )
+    try:
+        _call_git("branch", "-m", "master", "main", directory=directory)
+    except RuntimeError:
```

Comment:
@kcranston this doesn't seem to raise a Runtime error. Here is what happens. Can we capture the stdout (what is the correct type of output here) and check to see if it's fatal vs
```
# When main exists
(base) CIRES-EL-LM-020:my-assignment-template leahwasser$ git rev-parse --verify main
070a6cc0cca92cb3dc034fe1832529eb54fc7afb
# When a branch doesnt' exist  this is returned
(base) CIRES-EL-LM-020:my-assignment-template leahwasser$ git rev-parse --verify mainwe
fatal: Needed a single revision
(base) CIRES-EL-LM-020:my-assignment-template leahwasser$ git branch -m master main
fatal: A branch named 'main' already exists.
```

Reply:
I'd recommend `git show-ref --quiet --verify refs/heads/master`, which prints nothing and returns zero if the branch `master` exists, but not if a tag called `master` exists.

Reply:
thank you @jedbrown !! we will look into this!!